### PR TITLE
Chart title alignment

### PR DESCRIPTION
**Pull request by AlexVvx**
Related to [#882](https://github.com/plotly/plotly.js/issues/882)

This PR adds chart title alignment feature. The purpose of feature is to let users adjust the look of chart by using alignment. When there are lots of traces, then legend occupies much space and with center aligned title top margin looks ugly (details in https://github.com/ropensci/plotly/issues/812)

Title is an object in layout:

text - is title text
alignment - is string (top, bottom, left, right, middle default), small gap between border and title equal to half of text height
x,y - is numbers, manual position in pixels from left top

### CODE REVIEW

File: src/plot_api/subroutines.js [hunk 1]
```diff
@@ -410,16 +410,63 @@ function findCounterAxisLineWidth(gd, ax, subplotCounterLineWidth,
...
+                y: fullLayout._size.t / 2,
+                'text-anchor': 'middle'
+            };
+            break;
+        default:
+            attributes = {
+                x: fullLayout.title.x,
+                y: fullLayout.title.y,
+                'text-anchor': 'start',
+                'dominant-baseline': 'hanging'
```
Comment (etpinard) [inline src/plot_api/subroutines.js]: Hmm. Does `dominant-baseline` work in all browsers?

Comment (alexcjohnson) [reply src/plot_api/subroutines.js]: fraid not https://github.com/plotly/plotly.js/pull/2076#issuecomment-335318828

Comment (AlexVvx) [reply src/plot_api/subroutines.js]: Got it, will fix it together with other changes from https://github.com/plotly/plotly.js/issues/882

**Comment by ghost** [comment]
Why is it still an open issue? : o

**Comment by alexcjohnson** [comment]
@ewirth I don't recall the details at this point, but we ended up doing this in a slightly different way, in https://github.com/plotly/plotly.js/pull/3276 - so this PR, while useful for advancing the discussion, was ultimately not used directly. @AlexVvx can we close it?

**Comment by AlexVvx** [comment]
Yep, let's close

**Comment by ghost** [comment]
@AlexVvx Thanks for the notice. I think that it's still impossible to set the title to the bottom of a line figure. Or I haven't find it. It should be a easy thing like position: bottom :d

**Comment by alexcjohnson** [comment]
https://codepen.io/alexcjohnson/pen/NMdMbE?editors=1010

**Pull request closed without merging**